app/group.go

```diff
@@ -372,3 +396,26 @@ func (a *App) UserIsInAdminRoleGroup(userID, syncableID string, syncableType mod
 
 	return true, nil
 }
+
+// GetAllGroupsByTeam iterates all pages of groups associated to the given team and returns them all.
+func (a *App) GetAllGroupsByTeam(teamID string) ([]*model.Group, *model.AppError) {
+	batchCount := 1
+	page := 0
+	var teamGroups []*model.Group
+
+	for batchCount > 0 {
+		batch, err := a.Srv().Store.Group().GetGroupsByTeam(teamID, model.GroupSearchOpts{PageOpts: &model.PageOpts{Page: page, PerPage: 100}})
+		if err != nil {
+			return nil, err
+		}
+
+		batchCount = len(batch)
```

Comment:
Maybe I don't understand it fully, but: Does this mean it will stop if there are no groups in a page, even though that may be because there was no intersection (even though there may be more in latter pages)? In other words: we may have a batchCount of 0, but we're not done yet?